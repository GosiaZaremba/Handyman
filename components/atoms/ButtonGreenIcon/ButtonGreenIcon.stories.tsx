import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {ButtonGreenIcon} from './ButtonGreenIcon';
import {text} from '@storybook/addon-knobs';
import {action} from '@storybook/addon-actions';

storiesOf('atoms/ButtonGreenIcon', module)
  .addDecorator(story => story())
  .add('default', () => (
    <ButtonGreenIcon
      buttonText={text('button text', 'Button')}
      onPress={action('green button click')}
      iconName={'arrowright'}
    />
  ));
