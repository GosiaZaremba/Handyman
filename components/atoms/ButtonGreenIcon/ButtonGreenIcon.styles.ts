import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  gradient: {
    borderRadius: 10,
    width: '100%',
    height: 40,
    justifyContent: 'center',
    elevation: 8,
  },
  pressable: {
    justifyContent: 'space-evenly',
    alignItems: 'center',
    flexDirection: 'row',
  },
  pressed: {
    opacity: 0.8,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    flexDirection: 'row',
  },
  buttonText: {
    color: Colors.font.white,
    fontFamily: 'OpenSans-Bold',
  },
  icon: {
    fontSize: 18,
    color: Colors.font.white,
  },
});
