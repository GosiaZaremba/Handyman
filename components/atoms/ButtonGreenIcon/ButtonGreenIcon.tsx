import React from 'react';
import {Pressable, Text} from 'react-native';
import styles from './ButtonGreenIcon.styles';
import LinearGradient from 'react-native-linear-gradient';
import {Colors} from '../../../constants/colors';
import Icon from 'react-native-vector-icons/AntDesign';

export type Props = {
  buttonText: string;
  onPress: () => void;
  iconName: string;
  testID?: string;
  testOnly_pressed?: boolean;
};

export const ButtonGreenIcon: React.FC<Props> = ({
  buttonText,
  onPress,
  iconName,
  testID,
  testOnly_pressed,
}) => {
  return (
    <LinearGradient
      colors={[Colors.button.light, Colors.button.dark]}
      start={{x: 0, y: 0.5}}
      end={{x: 1, y: 0.5}}
      style={styles.gradient}>
      <Pressable
        testID={testID}
        testOnly_pressed={testOnly_pressed}
        style={({pressed}) => (pressed ? styles.pressed : styles.pressable)}
        onPress={onPress}>
        <Text style={styles.buttonText}>{buttonText}</Text>
        <Icon name={iconName} style={styles.icon} />
      </Pressable>
    </LinearGradient>
  );
};
