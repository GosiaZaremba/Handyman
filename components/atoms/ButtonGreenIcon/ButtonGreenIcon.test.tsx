import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {ButtonGreenIcon} from './ButtonGreenIcon';

const mockOnPress = jest.fn();

describe('Atom buttonGreenIcon', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <ButtonGreenIcon
          onPress={mockOnPress}
          buttonText={'Button Green'}
          testID="atom-buttonGreen"
          testOnly_pressed={true}
          iconName={'arrowright'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure text is passed correctly', () => {
    const {getByTestId} = render(
      <ButtonGreenIcon
        onPress={mockOnPress}
        buttonText={'Button Green Icon'}
        testID="atom-buttonGreenIcon"
        testOnly_pressed={false}
        iconName={'arrowright'}
      />,
    );
    const addPhotoButton = getByTestId('atom-buttonGreenIcon');
    expect(addPhotoButton).toHaveTextContent('Button Green Icon');
  });

  test('ensure onPress works correctly', () => {
    const {getByTestId} = render(
      <ButtonGreenIcon
        onPress={mockOnPress}
        buttonText={'Button Green Icon'}
        testID="atom-buttonGreenIcon"
        testOnly_pressed={false}
        iconName={'arrowright'}
      />,
    );
    const buttonGreenIcon = getByTestId('atom-buttonGreenIcon');
    fireEvent.press(buttonGreenIcon);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(buttonGreenIcon.props.style.opacity).toBe(undefined);
  });

  test('ensure pressable pressed works correctly', () => {
    const {getByTestId} = render(
      <ButtonGreenIcon
        onPress={mockOnPress}
        buttonText={'Button Green Icon'}
        testID="atom-buttonGreenIcon"
        testOnly_pressed={true}
        iconName={'arrowright'}
      />,
    );
    const buttonGreenIcon = getByTestId('atom-buttonGreenIcon');
    fireEvent.press(buttonGreenIcon);
    expect(buttonGreenIcon.props.style.opacity).toBe(0.8);
  });
});
