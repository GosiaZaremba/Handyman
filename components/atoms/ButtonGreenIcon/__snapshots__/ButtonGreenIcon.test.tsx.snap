// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom buttonGreenIcon Component Snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278363496,
      4285643865,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.5,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  style={
    Object {
      "borderRadius": 10,
      "elevation": 8,
      "height": 40,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "opacity": 0.8,
      }
    }
    testID="atom-buttonGreen"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Bold",
        }
      }
    >
      Button Green
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î˜›
    </Text>
  </View>
</BVLinearGradient>
`;
